src/main/scala/scorex/transaction/assets/MassTransferTransaction.scala

```diff
@@ -0,0 +1,126 @@
+          val amount = Longs.fromByteArray(bytes.slice(ofs, ofs + 8))
+          (Right[ValidationError, (AddressOrAlias, Long)]((addr, amount)), ofs + 8)
+        case Left(e) => (Left(e), offset)
+      }
+    }
+    val recipientsList: List[(Validation[(AddressOrAlias, Long)], Int)] =
+      List.iterate(readRecipient(s0 + 2), recipientCount) { case (_, offset) => readRecipient(offset) }
+    val recipientsEi: Validation[List[(AddressOrAlias, Long)]] = recipientsList.map { case (ei, _) => ei }.sequence
+
+    val s1 = recipientsList.lastOption.map(_._2).getOrElse(s0 + 2)
+    val tx: Validation[MassTransferTransaction] = for {
+      recipients <- recipientsEi
+      timestamp = Longs.fromByteArray(bytes.slice(s1, s1 + 8))
+      feeAmount = Longs.fromByteArray(bytes.slice(s1 + 8, s1 + 16))
+      (attachment, s2) = Deser.parseArraySize(bytes, s1 + 16)
+      signature = ByteStr(bytes.slice(s2, s2 + SignatureLength))
+      mtt <- MassTransferTransaction.create(assetIdOpt.map(ByteStr(_)), sender, recipients, timestamp, feeAmount, attachment, signature)
+    } yield mtt
+    tx.fold(left => Failure(new Exception(left.toString)), right => Success(right))
+  }.flatten
+
+  def create(assetId: Option[AssetId],
+             sender: PublicKeyAccount,
+             recipients: List[(AddressOrAlias, Long)],
+             timestamp: Long,
+             feeAmount: Long,
+             attachment: Array[Byte],
+             signature: ByteStr): Either[ValidationError, MassTransferTransaction] = {
+    Try { recipients.map(_._2).fold(feeAmount)(Math.addExact) }.fold(
+      ex => Left(ValidationError.OverflowError),
+      totalAmount =>
+        if (recipients.lengthCompare(MaxTransferCount) > 0) {
+          Left(ValidationError.GenericError(s"Number of recipients is greater than $MaxTransferCount"))
+        } else if (recipients.exists(_._2 < 0)) {
+          Left(ValidationError.GenericError("One of the transfers has negative value"))
+        } else if (attachment.length > TransferTransaction.MaxAttachmentSize) {
+          Left(ValidationError.TooBigArray)
+        } else if (feeAmount <= 0) {
+          Left(ValidationError.InsufficientFee)
+        } else {
+          Right(MassTransferTransaction(assetId, sender, recipients, timestamp, feeAmount, attachment, signature))
+        }
+    )
+  }
+
+  def create(assetId: Option[AssetId],
+             sender: PrivateKeyAccount,
+             recipients: List[(AddressOrAlias, Long)],
+             timestamp: Long,
+             feeAmount: Long,
+             attachment: Array[Byte]): Either[ValidationError, MassTransferTransaction] = {
+    create(assetId, sender, recipients, timestamp, feeAmount, attachment, ByteStr.empty).right.map { unsigned =>
+      unsigned.copy(signature = ByteStr(EllipticCurveImpl.sign(sender, unsigned.toSign())))
+    }
+  }
+
+  def processRecipientsWith[T, R](recipients: List[(T, Long)])(f: (T, Long) => Validation[R]): Validation[List[R]] =
```

Comment:
What is `T`? `AddressOrAlias` ?
Isn't it just `.traverse` ?


`def traverse[G[_]: Applicative, B](f : C => G[B]): G[F[B]]`